apm-agent-plugins/apm-jdbc-plugin/src/test/java/co/elastic/apm/agent/jdbc/helper/ConnectionMetaDataTest.java

```diff
@@ -33,297 +35,333 @@ class ConnectionMetaDataTest {
+            "(address_list=(address=(protocol=tcp)(port=1521)(host=oraHost)))" +
+            "(connect_data=(SERVICE_NAME=service_name)(INSTANCE_NAME=instance_name)))", "oracle", "orahost", 1521, "instance_name");
     }
 
     @Test
     void testPostgresql() {
         // https://jdbc.postgresql.org/documentation/head/connect.html
-        testUrl("jdbc:postgresql://myhost:666/database", "postgresql", "myhost", 666);
-        testUrl("jdbc:postgresql://myhost/database", "postgresql", "myhost", 5432);
-        testUrl("jdbc:postgresql://myhost:666/", "postgresql", "myhost", 666);
-        testUrl("jdbc:postgresql://myhost/", "postgresql", "myhost", 5432);
-        testUrl("jdbc:postgresql://127.0.0.1/", "postgresql", "127.0.0.1", 5432);
-        testUrl("jdbc:postgresql://::1/", "postgresql", "::1", 5432);
-        testUrl("jdbc:postgresql://[::1]/", "postgresql", "[::1]", 5432);
-        testUrl("jdbc:postgresql://[::1]:666/", "postgresql", "[::1]", 666);
-
-        testUrl("jdbc:postgresql:database", "postgresql", "localhost", -1);
-        testUrl("jdbc:postgresql:/", "postgresql", "localhost", -1);
+        testUrl("jdbc:postgresql://myhost:666/database", "postgresql", "myhost", 666, "database");
+        testUrl("jdbc:postgresql://myhost/database", "postgresql", "myhost", 5432, "database");
+        testUrl("jdbc:postgresql://myhost:666/", "postgresql", "myhost", 666, null);
+        testUrl("jdbc:postgresql://myhost/", "postgresql", "myhost", 5432, null);
+        testUrl("jdbc:postgresql://127.0.0.1/", "postgresql", "127.0.0.1", 5432, null);
+        testUrl("jdbc:postgresql://::1/", "postgresql", "::1", 5432, null);
+        testUrl("jdbc:postgresql://[::1]/", "postgresql", "[::1]", 5432, null);
+        testUrl("jdbc:postgresql://[::1]:666/", "postgresql", "[::1]", 666, null);
+
+        testUrl("jdbc:postgresql:database", "postgresql", "localhost", 5432, "database");
+        testUrl("jdbc:postgresql:/", "postgresql", "localhost", 5432, null);
     }
 
     @Test
     void testIngress() {
         // https://docs.actian.com/ingres/11.0/index.html#page/QuickStart_Win/5._Connecting_to_Ingres_Using_JDBC.htm
-        testUrl("jdbc:ingres://myhost:II7/testdb;UID=userid;PWD=password", "ingres", "myhost", -1);
-        testUrl("jdbc:ingres://localhost:II7/testdb;UID=userid;PWD=password", "ingres", "localhost", -1);
-        testUrl("jdbc:ingres://localhost:567/testdb;UID=userid;PWD=password", "ingres", "localhost", 567);
+        testUrl("jdbc:ingres://myhost:II7/testdb;UID=userid;PWD=password", "ingres", "myhost", -1, "testdb");
+        testUrl("jdbc:ingres://localhost:II7/testdb;UID=userid;PWD=password", "ingres", "localhost", -1, "testdb");
+        testUrl("jdbc:ingres://localhost:567/testdb;UID=userid;PWD=password", "ingres", "localhost", 567, "testdb");
     }
 
     @Test
     void testMysql() {
         // https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-jdbc-url-format.html
         // Single host:
```

Comment:
`testUrl("jdbc:p6spy:mysql://myhost:666/database", "mysql", "myhost", 666, "database");`
would fail. I know it's an edge case and why are they using p6spy when they're using our agent, but it's possible

Reply:
I'm not very familiar with P6Spy implementation and need to test that to be sure, but it would likely behave exactly like the Testcontainers JDBC URLs : https://www.testcontainers.org/modules/databases/jdbc/#jdbc-url-examples (we are actually using them in `JdbcDbIT`) where the URL that we capture is provided by a call to `java.sql.Connection#getMetaData`, not the URL that is provided through the `java.sql.DriverManager#getConnection(java.lang.String, java.util.Properties)`

Reply:
I just manually tested within the opbeans java app, and the agent is able to properly capture the `h2` when setting `spring.datasource.url=jdbc:p6spy:h2:mem:opbeans` in `application.properties` so I think we are good to go, but it's definitely something that we will have to change if we start to instrument at the driver level.

Reply:
excellent!